### ISSUE TITLE
Incompatibility with Music Thing Turing Machine?

### ISSUE THREAD

**Issue by grimnebluna** [arirusso/twigs#1]
hi,

i've been using twigs with turing machine. the clock pulses are too short or too weak for turing machine to react sometimes. the LEDs only blink very weakly.

any suggestions?

cheers.

**Comment by arirusso** [comment]
@grimnebluna Sorry, I need to fix my github notifications- I only just noticed this

Thanks for posting

I've never owned a Turing Machine so bear with me.  I'd like to ask a few questions about your setup in order to resolve this:

Are you using the original version or the mk2 Turing Machine? Are you using the Pulses Expander? 

From what I've read, Turing Machine may output whatever length/voltage pulse you input into it depending on the version.  If that's the case is it possible that the module(s) that's being triggered by Turing Machine only accepts gate and not trigger pulses?

The length/voltage that Twigs outputs is the same as Branches.  Did you ever try the same configuration but with Branches driving Turing Machine?

Thanks again

**Comment by grimnebluna** [comment]
@arirusso 
thank anyway for your reply!! :)

I was using the MK1, but as of today I have also the MK2 in my setup. Yes, both with pulses expander.

The shortened pulses already look weird on the Pulses module (LED flashing very weakly). It almost looks as if the incoming V are too low. However on other modules it worked .. May be that Turing machines MK1 only accepts gate. I will check that.

I will try the same configuration with the original branches software and get back to you!

cheers!

**Comment by arirusso** [comment]
Yeah give that a try and also try switching what module you're sending out to

The outgoing voltage from Branches/Twigs is +5V.  That's something that's determined by the hardware, not the firmware

It might be possible to make the pulses longer coming out of Twigs but I'd like to figure out what's going on exactly before I do anything. 

I have to say that I've tested Twigs with dozens of modules and it's worked as expected. I suspect that either TM or whatever module is the final destination for these pulses is expecting something more like gate than trig

We'll figure it out...

Thank you

**Comment by arirusso** [comment]
@grimnebluna

Yeah give that a try and also try switching what module you're sending out to, just as a test

The outgoing voltage from Branches/Twigs is +5V.  That's something that's determined by the hardware, not the firmware

It might be possible to make the pulses longer coming out of Twigs but I'd like to figure out what's going on exactly before I do anything

I suspect that either TM or whatever module is the final destination for these pulses is expecting something more like gate than trig.   I've tested it with a lot of trig input modules and not a had a problem

We'll figure it out...

Thank you

**Comment by ukyzky** [comment]
@arirusso 
Hi, thanks for your great firmware!

@grimnebluna 
I fix the too short trigger length problem.
https://github.com/ukyzky/twigs/blob/triggerextended/README.md
https://github.com/ukyzky/twigs/blob/triggerextended/releases/twigs-1.0.1-triggerextended.tar.gz

In my test, Mutable Instruments Branches hardware with twigs original firmware trigger output and Make Noise Rene Classic module clock input have sometimes lack of clock trigger recognition.
And on my fixed custom firmware, the lack is disappear.

I don't test with turing machine. So I can't guarantee the problem fixed all.
But please try my firmware if you can.

(edit) at now, I replaced the release file for more increase trigger length (become 2.56ms).

**Comment by arirusso** [comment]
Above fix is in https://github.com/arirusso/twigs/releases/tag/v1.0.2
